pep-0723.rst

```diff
@@ -0,0 +1,185 @@
+PEP: 723
+Title: Embedding metadata in single-file scripts
+Author: Ofek Lev <ofekmeister@gmail.com>
+PEP-Delegate: Brett Cannon <brett@python.org>
+Discussions-To: https://discuss.python.org/t/29905
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Content-Type: text/x-rst
+Created: 04-Aug-2023
+Post-History: `19-Jul-2023 <https://discuss.python.org/t/29905>`__
+
+
+Abstract
+========
+
+This PEP specifies a format for defining metadata about single-file Python
+scripts, such as their runtime dependencies. This PEP merely extends
+:pep:`722` with an alternative format and therefore elides describing the
+motivation for such a proposal.
+
+Specification
+=============
+
+This PEP chooses to follow the latest developments of other modern packaging
+ecosystems (namely `Rust <https://github.com/rust-lang/rfcs/blob/master/text/3424-cargo-script.md>`__
+and `Go <https://github.com/erning/gorun>`__) by embedding the existing
+standard :pep:`621` that is used to describe projects.
+
+Any Python script may assign a variable named ``__pyproject__`` to a multi-line
+double-quoted string containing a valid TOML document. This document contains
+what you would expect to see in a ``pyproject.toml`` file, with the following
```

Comment:
I don't think we can assume what the reader would expect. For example, someone might expect only project metadata, or only tools config, or both...

Can we reword this a bit?

Reply:
I agree with @hugovk. The PEP should be explicit about what is and is not allowed. If it is allowed to include anything that's valid in `pyproject.toml`, that should be explicit (and the PEP then needs to address the implications of that).